### ISSUE TITLE
[BUG] `server_session` example does not update plot title

### ISSUE THREAD

**Issue by carve11** [bokeh/bokeh#13172]
### Software versions

Python version      :  3.9.12 (main, Mar 26 2022, 15:52:10) 
IPython version     :  (not installed)
Tornado version     :  6.3.2
Bokeh version       :  3.1.1
BokehJS static path :  /Users/xxx/programmering/python/bokeh/server/flask_embed/.venv/lib/python3.9/site-packages/bokeh/server/static
node.js version     :  v16.14.2
npm version         :  8.5.0
Operating system    :  macOS-12.5-x86_64-i386-64bit

### Expected behavior

When I run the [`server_session` example](examples/output/apis/server_session/serve.py) I had expected the title of the plot to get updated based on the code:
https://github.com/bokeh/bokeh/blob/cb2de9ea4b92e54f2fe4b3b4a460c6d49c762f53/examples/output/apis/server_session/serve.py#L17

### Observed behavior

Title of plot is not updated. The plot itself in `bokeh_app.py` is not initialized with a title text, but I have also tested that and do not observe that the text gets updated

### Example code

```Python
The example is located in bokeh/examples/output/apis/server_session/
```

### Screenshots

<img width="628" alt="Skærmbillede 2023-05-30 kl  18 24 51" src="https://github.com/bokeh/bokeh/assets/33606696/b61c958e-12c9-44d5-b1e2-0dea3794e7d9">

**Comment by huijeong1015** [comment]
I think it's missing `push_session` function in `bokeh/examples/output/apis/server_session/serve.py`.

When I call `push_session(session_id=session.id, document=session.document, url=app_url)` after code:
https://github.com/bokeh/bokeh/blob/cb2de9ea4b92e54f2fe4b3b4a460c6d49c762f53/examples/output/apis/server_session/serve.py#L17
I get an expected result:
![image](https://github.com/bokeh/bokeh/assets/64668622/47f2ee0c-f7f3-4a32-b0a1-5c6b25213a0e)

My understanding is that without the `push_session`, the Bokeh server session does not get updated, so when it generates a script using `server_session`, it does not reflect the changes in the plot.

However, I'm unsure if this is the correct way to resolve this issue. I'll make a PR for now with the changes.

**Comment by bryevdv** [comment]
Well, that did not used to be necessary, and in principle it should not be necessary either. Updating a model property here should be just like updating a model property in an `on_change` callback, i.e. setting the value should automatically trigger the session to sync on the server. 

However, it's possible there is just an unavoidable race condition here, and we were only just happening to get "lucky" in the past. I'm not sure I would oppose an explicit synchronization, however I think that could be accomplished via `session.push()` rather than having to import `push_session`. 

cc @bokeh/dev for any comments

**Comment by bryevdv** [comment]
One thing to note is that in the past, the`__exit__` for `ClientSession` used to follow a codepath that went through a call to `_force_roundtrip` and that does not seem to be the case any longer. I am not sure that would explain the different behavior, though, since the call to `server_session` and the template render were still inside the context manager in the past. But maybe *that* was the race condition, and automatic property synchronization has never been occurring.

**Comment by huijeong1015** [comment]
> One thing to note is that in the past, the`__exit__` for `ClientSession` used to follow a codepath that went through a call to `_force_roundtrip` and that does not seem to be the case any longer. I am not sure that would explain the different behavior, though, since the call to `server_session` and the template render were still inside the context manager in the past. But maybe _that_ was the race condition, and automatic property synchronization has never been occurring.

I think if we are expecting `server_session` to reflect the changes of the current session's document, there needs to be some type of "pushing" inside the function. Currently, it only retrieves information from the existing Bokeh session without any updates.

**Comment by bryevdv** [comment]
> Currently, it only retrieves information from the existing Bokeh session without any updates.

Right, but many things in Bokeh auto-synchronize. E.g. there is no "push" back to the client in any `on_change` callback in any Bokeh server app, by design. The metaclass descriptors for the Bokeh model objects handle detecting property updates and automatically syncing those across any connection. That was definitely how this used to work as well, and how it was envisioned to work. i.e. the current example code has definitely *stopped working*, from its previous working state. 

But I am willing to consider that situation is not entirely analogous, that the previous working state was fragile, and that we should just adopt and explicit push for Python <--> Python session updates, going forward. But I'd like some other input from @bokeh/core first.
